spec.md

```diff
@@ -230,6 +245,39 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Public Modules
+
+A public module is accessible by `Circuit`s other than the `Circuit` it is
+defined in. Unless otherwise specified, all modules inside of a `Circuit` are
+private and cannot be accessed by any other `Circuit`.
+
+A public module has fixed ports.
+
+A module can be marked as public like so:
+
+```firrtl
+circuit FooCircuit :
+   module Foo is public :
+      input foo: UInt
+      output bar: UInt
+      ; ...
+```
+
+A public module can be instantiated like so:
+
+```firrtl
+extcircuit FooCircuit :
+  extmodule Foo :
+      input foo: UInt
+      output bar: UInt
+      defname = VerilogName
```

Comment:
again this is just a suggestion/placeholder for "other stuff we might eventually add that are not just ports", we should probably delete this `defname` thing for now.

Reply:
I added these in the style of other "placeholder" stuff in the `ExtModule` example... do you think it's better to leave the contents empty for now? Like just

```
extModule Foo :
  ; ...
```